extensions/recents/src/recents.tsx

```diff
@@ -49,6 +54,7 @@ const getRecents = async (scope: string | undefined, callback: (result: Spotligh
       query: queryParts.query,
       directories: queryParts.directories,
       attributes: ["kMDItemDisplayName", "kMDItemKind", "kMDItemLastUsedDate"],
+      limit,
```

Comment:
```suggestion
      limit || 1000,
```
If you delete the preference value it's blank and might cause the heap size error again 😊

Reply:
done! good spot!